CRM/Core/Payment/OmnipayMultiProcessor.php

```diff
@@ -732,7 +736,16 @@ public function processPaymentNotification($params) {
     $response = NULL;
     try {
       if ($this->gateway->supportsAcceptNotification()) {
-        $response = $this->gateway->acceptNotification($params)->send();
+        $response = $this->gateway->acceptNotification($params);
+
+        $transactionId = $response->getTransactionId();
+        $transactionReference = civicrm_api3('Contribution', 'getvalue', ['id' => $transactionId, 'return' => 'trxn_id']);
```

Comment:
I think getvalue will return errors where empty - I think here & above it would be safer with some empty handling for other gateways